packages/installer/cos.sh

```diff
@@ -1242,11 +1395,19 @@ install() {
     # We want to find the upgrade channel if, no ISO url is supplied and:
     # 1: We aren't booting from LiveCD - the rootfs that we are going to install must be downloaded from somewhere
     # 2: If we specify directly an image to install
-    if ! is_booting_from_live && [ -z "$COS_INSTALL_ISO_URL" ] || [ -n "$COS_IMAGE" ] && [ -z "$COS_INSTALL_ISO_URL" ]; then
+    if ! is_booting_from_live && [ -z "$_COS_INSTALL_ISO_URL" ] || [ -n "$_COS_IMAGE" ] && [ -z "$_COS_INSTALL_ISO_URL" ]; then
         find_upgrade_channel
+    else
+      # Beforehand the config was only loaded if the path above was set, so we loaded some vars needed for the upgrade
+      # But now the config is loaded on start, so we need to unset some of those loaded vars if we dont need them, otherwise
+      # we can wrongly install stuff
+      unset _UPGRADE_IMAGE
+      unset _RECOVERY_IMAGE
+      # Set channel upgrades to default value
+      _CHANNEL_UPGRADES=true
```

Comment:
mmh maybe this is not necessary, as it makes sense only if we have UPGRADE_IMAGE set

Reply:
indeed, no matter if this is set or not in this path